install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -454,6 +454,9 @@ data:
 {{- end }}
 {{- if and .Values.ipMasqAgent .Values.ipMasqAgent.enabled }}
   enable-ip-masq-agent: "true"
+  config:
```

Comment:
Have you tested your change? I don't see how this `config` can be propagated as the ip-masq-agent setting to cilium-agent.

Reply:
uh i miss something..
```
- name: ip-masq-agent
        configMap:
          name: cilium-config  # change it to cilium-config
          optional: true
          items:
          - key: config
            path: ip-masq-agent
```
I made some test for helm install, but the agent pod failed to init, No useful information could be found from either `kubelet` or `kubectl describe`

```
[root@master cilium]# helm install cilium . --set image.tag=v1.12.0-rc2  --set operator.image.tag=v1.12.0-rc2
NAME: cilium
LAST DEPLOYED: Fri Jun 10 20:01:55 2022
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
You have successfully installed Cilium with Hubble.

Your release version is 1.11.90.

For any further help, visit https://docs.cilium.io/en/v1.11/gettinghelp
[root@master cilium]# kubectl get po
NAME                               READY   STATUS       RESTARTS     AGE
cilium-cfvsk                       0/1     Init:Error   0            4s
cilium-gt2q9                       0/1     Init:Error   1 (3s ago)   4s
cilium-operator-84cbdd8b5c-dbr87   1/1     Running      0            4s
cilium-operator-84cbdd8b5c-jq7pf   1/1     Running      0            4s
```
```
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  33s                default-scheduler  Successfully assigned default/cilium-cfvsk to master
  Normal   Pulled     32s                kubelet            Successfully pulled image "quay.io/cilium/cilium:v1.12.0-rc2" in 987.115567ms
  Normal   Pulled     30s                kubelet            Successfully pulled image "quay.io/cilium/cilium:v1.12.0-rc2" in 1.007922246s
  Normal   Pulling    13s (x3 over 33s)  kubelet            Pulling image "quay.io/cilium/cilium:v1.12.0-rc2"
  Normal   Created    12s (x3 over 32s)  kubelet            Created container mount-cgroup
  Normal   Started    12s (x3 over 32s)  kubelet            Started container mount-cgroup
  Warning  BackOff    12s (x3 over 29s)  kubelet            Back-off restarting failed container
  Normal   Pulled     12s                kubelet            Successfully pulled image "quay.io/cilium/cilium:v1.12.0-rc2" in 1.064636306s
```
This seems to no matter with `ipMasqAgent`, enable it  or disable  has the same result..